### PR TITLE
🌗 Enable windows QNN e2e test

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes
1. Enable windows QNN e2e test
2. Added use conda path into windows ci test if it is for qnn. This is a workaround for this kind of issue only ci pipeline: https://community.anaconda.cloud/t/cant-install-failed-to-extract-packages-error/62687
3. Update pwsh Join-path to back-compat the version of pwsh <= 5.x, 

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: examples/test/test_qnn_tooklit.py [hunk 1]
```diff
@@ -11,35 +13,54 @@
...
+    elif platform.system() == "Linux":
+        run_subprocess(cmd=f"unzip {download_path} -d {str(target_path)}", check=True)
     return target_path
 
 
 @pytest.fixture(scope="module", autouse=True)
 def setup():
     """Setups any state specific to the execution of the given module."""
+    tmp_dir = tempfile.TemporaryDirectory()
+    tmp_path = Path(tmp_dir.name)
```
Comment (guotuofeng) [inline examples/test/test_qnn_tooklit.py:44]: use tmp_path fixture instead?

Comment (trajepl) [reply examples/test/test_qnn_tooklit.py:44]: https://stackoverflow.com/questions/62335029/how-do-i-use-tmpdir-with-my-pytest-fixture 
Seems can not use it here as the tmp_dir is passed into another function of `download_qnn_sdk`.

That leads the error: https://aiinfra.visualstudio.com/PublicPackages/_build/results?buildId=401411&view=logs&j=4f5f4a75-935b-5877-7f37-0e73f78960b0&t=b01a3c21-1243-5721-75b4-0ba5e3889b9d

File: .azure_pipelines/job_templates/olive-example-template.yaml [hunk 2]
```diff
@@ -43,6 +43,10 @@ jobs:
       displayName: Set exampleRequirements
       condition: eq(variables['exampleRequirements'], '')
 
+    - powershell: Write-Host "##vso[task.prependpath]$env:CONDA\Scripts"
```
Comment (guotuofeng) [inline .azure_pipelines/job_templates/olive-example-template.yaml:46]: >powershell [](http://example.com/codeflow?start=6&length=10)

will linux call this one?

File: .azure_pipelines/job_templates/olive-example-template.yaml [hunk 3]
```diff
@@ -43,6 +43,10 @@ jobs:
...
+      displayName: Add conda to PATH
+      condition: and(eq(parameters.name, 'Windows_CI'), eq(variables['exampleFolder'], 'mobilenet_qnn_qualcomm_npu'))
```
Comment (guotuofeng) [inline .azure_pipelines/job_templates/olive-example-template.yaml:48]: >Windows_CI [](http://example.com/codeflow?start=42&length=10)

there is a concept named "windows" in test template. do we need introduce to example to keep consistent?

File: examples/test/test_qnn_tooklit.py (hunk 1 above)
Comment (guotuofeng) [inline examples/test/test_qnn_tooklit.py:44]: I suppose [https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/consider-using-with.html](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/consider-using-with.html) will be raised?

Comment (guotuofeng) [reply examples/test/test_qnn_tooklit.py:44]: that's the reason I suggest to use [https://docs.pytest.org/en/7.4.x/how-to/tmp_path.html](https://docs.pytest.org/en/7.4.x/how-to/tmp_path.html)

File: .azure_pipelines/job_templates/olive-example-template.yaml (hunk 3 above)
Comment (trajepl) [reply .azure_pipelines/job_templates/olive-example-template.yaml]: Updated.

File: .azure_pipelines/job_templates/olive-example-template.yaml [hunk 4]
```diff
@@ -43,6 +43,10 @@ jobs:
       displayName: Set exampleRequirements
       condition: eq(variables['exampleRequirements'], '')
 
+    - powershell: Write-Host "##vso[task.prependpath]$env:CONDA\Scripts"
+      displayName: Add conda to PATH
+      condition: and(eq('${{ parameters.windows }}', true), eq(variables['exampleFolder'], 'mobilenet_qnn_qualcomm_npu'))
```
Comment (guotuofeng) [inline .azure_pipelines/job_templates/olive-example-template.yaml:48]: the parameters need to be passed to the template.

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**